api/v1alpha1/servicedefaults_types.go

```diff
@@ -0,0 +1,143 @@
+	Items           []ServiceDefaults `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&ServiceDefaults{}, &ServiceDefaultsList{})
+}
+
+// ToConsul converts the entry into it's Consul equivalent struct.
+func (s *ServiceDefaults) ToConsul() *capi.ServiceConfigEntry {
+	return &capi.ServiceConfigEntry{
+		Kind: capi.ServiceDefaults,
+		Name: s.Name,
+		//Namespace: s.Namespace, // todo: don't set this unless enterprise
+		Protocol:    s.Spec.Protocol,
+		MeshGateway: s.Spec.MeshGateway.ToConsul(),
+		Expose:      s.Spec.Expose.ToConsul(),
+		ExternalSNI: s.Spec.ExternalSNI,
+	}
+}
+
+// MatchesConsul returns true if entry has the same config as this struct.
+func (s *ServiceDefaults) MatchesConsul(entry *capi.ServiceConfigEntry) bool {
+	return s.Name == entry.GetName() &&
+		s.Spec.Protocol == entry.Protocol &&
+		s.Spec.MeshGateway.Mode == string(entry.MeshGateway.Mode) &&
+		s.Spec.Expose.Matches(entry.Expose) &&
+		s.Spec.ExternalSNI == entry.ExternalSNI
+}
+
+// ExposeConfig describes HTTP paths to expose through Envoy outside of Connect.
+// Users can expose individual paths and/or all HTTP/GRPC paths for checks.
+type ExposeConfig struct {
+	// Checks defines whether paths associated with Consul checks will be exposed.
+	// This flag triggers exposing all HTTP and GRPC check paths registered for the service.
+	Checks bool `json:"checks,omitempty"`
+
+	// Paths is the list of paths exposed through the proxy.
+	Paths []ExposePath `json:"paths,omitempty"`
+}
+
+type ExposePath struct {
+	// ListenerPort defines the port of the proxy's listener for exposed paths.
+	ListenerPort int `json:"listenerPort,omitempty"`
+
+	// Path is the path to expose through the proxy, ie. "/metrics."
+	Path string `json:"path,omitempty"`
+
+	// LocalPathPort is the port that the service is listening on for the given path.
+	LocalPathPort int `json:"localPathPort,omitempty"`
+
+	// Protocol describes the upstream's service protocol.
+	// Valid values are "http" and "http2", defaults to "http"
+	Protocol string `json:"protocol,omitempty"`
+}
+
+//Matches returns true if the expose config of the entry is the same as the struct
+func (e ExposeConfig) Matches(expose capi.ExposeConfig) bool {
+	if e.Checks != expose.Checks {
+		return false
+	}
+
+	if len(e.Paths) != len(expose.Paths) {
+		return false
+	}
+
+	for _, path := range e.Paths {
+		found := false
+		for _, entryPath := range expose.Paths {
+			if path.Protocol == entryPath.Protocol &&
+				path.Path == entryPath.Path &&
+				path.ListenerPort == entryPath.ListenerPort &&
+				path.LocalPathPort == entryPath.LocalPathPort {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+	return true
+}
+
+//ToConsul returns the ExposeConfig for the entry
+func (e ExposeConfig) ToConsul() capi.ExposeConfig {
+	return capi.ExposeConfig{
+		Checks: e.Checks,
+		Paths:  e.parseExposePath(),
+	}
+}
+
+func (e ExposeConfig) parseExposePath() []capi.ExposePath {
```

Comment:
I'd inline this function into `ToConsul()`. The parse part of the name seems weird to me.